students/ericstreit/lessons/lesson09/assignment/database.py

```diff
@@ -0,0 +1,238 @@
+"""
+#Lesson10
+Metaprogramming Timing
+Special Thanks to:
+https://www.blog.pythonlibrary.org/2016/05/24/python-101-an-intro-to-benchmarking-your-code/
+"""
+
+#!/usr/bin/env python3
+#import stuff here
+from pymongo import MongoClient
+import logging
+import csv
+import time
+
+
+#global variables here
+logging.basicConfig(level=logging.ERROR)
+#define function and classes
+
+def timerfunc(func):
+    """ a timer decorator """
+
+    def function_timer(*args, **kwargs):
+        """ nested function for timing functions """
+
+        start = time.time()
+        value = func(*args, **kwargs)
+        end = time.time()
+        runtime = end - start
+        print(f' The runtime for {func.__name__} took {runtime} seconds to complete')
+        return value
+    return function_timer
```

Comment:
Nice decorator. :) Not sure this is really what assignment 10 was going for, since we already did decorators, but more important to learn decorators then metaclasses or other meta constructs, so I'll give it to you. ;-)